Cabal/src/Distribution/Types/ParStrat.hs

```diff
@@ -0,0 +1,9 @@
+module Distribution.Types.ParStrat where
+
+data ParStrat = UseSem String Int | Old (Maybe Int) | Serial
```

Comment:
I think you should rename the `Old` constructor and add some documentation, e.g.:

```suggestion
-- | How to control parallelism, e.g. a fixed number of jobs or by using a system semaphore.
data ParStrat
  -- | Compile in parallel with the given number of jobs (`-jN` or `-j`).
  = NumJobs (Maybe Int)
  -- | `--semaphore`: use a system semaphore to control parallelism.
  | UseSem String Int
  -- | No parallelism (neither `-j` nor `--semaphore`).
  | Serial
```